src/main/RemoteRequest.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * RemoteRequest is used to for http requests on a remote server which can be
+ * fetched in chunks, e.g. using a Range request.
+ * @flow
+ */
+'use strict';
+
+import Q from 'q';
+
+type Chunk = {
+  start: number;
+  stop: number;
+  buffer: Object; // TODO: more efficient data store
```

Comment:
pileup.js has always drawn a distinction between data classes and source classes:
https://github.com/hammerlab/pileup.js/blob/master/DESIGN.md

The data classes (e.g. `bam.js`) are responsible for issuing network requests and parsing bytes into structures. The source classes (e.g. `BamDataSource.js`) are responsible for caching features in a range (a `ContigInterval`) and exposing a synchronous API.

This class might be more like a source class than a data class. I think [`GA4GHDataSource`][1] is a close analogue to what you're doing here.

[1]: https://github.com/hammerlab/pileup.js/blob/55dc2213a9ccf6bdb336f402a42e0632776f8e7c/src/main/sources/GA4GHDataSource.js